### PR TITLE
Reject Prescription Creation if Medicine already Prescribed and is active

### PR DESCRIPTION
**Pull request by rithviknishad**
## Proposed Changes

- RReject Prescription Creation if Medicine already Prescribed and is active

### Associated Issue

- Required by coronasafe/care_fe#6085

image

## Merge Checklist

- [x] Tests added/fixed
- [ ] Update docs in `/docs`
- [x] Linting Complete
- [ ] Any other necessary step

_*Only PR's with test cases included and passing lint and test pipelines will be reviewed*_

@coronasafe/care-backend-maintainers @coronasafe/care-backend-admins

### CODE REVIEW

File: care/facility/api/serializers/prescription.py [hunk 1]
```diff
@@ -57,6 +62,25 @@ def validate(self, attrs):
...
+        if not self.instance:
+            consultation_obj = get_object_or_404(
+                PatientConsultation,
+                external_id=self.context["request"].parser_context["kwargs"][
+                    "consultation_external_id"
+                ],
+            )
+
+            if Prescription.objects.filter(
+                consultation=consultation_obj,
```
Comment (Ashesh3) [inline care/facility/api/serializers/prescription.py:74]: Can we not query the Prescription object directly via `consultation__external_id=self.context["request"].parser_context["kwargs"]["consultation_external_id"]`? That'd save a db query fetching the PatientConsultation object.

Comment (rithviknishad) [reply care/facility/api/serializers/prescription.py]: Done!

File: care/facility/api/serializers/prescription.py [hunk 2]
```diff
@@ -57,6 +57,20 @@ def validate(self, attrs):
...
+            if Prescription.objects.filter(
+                consultation__external_id=self.context["request"].parser_context[
+                    "kwargs"
+                ]["consultation_external_id"],
+                medicine=attrs["medicine"],
+                discontinued=False,
+            ).exists():
+                raise serializers.ValidationError(
+                    {
+                        "medicine": "This medicine is already prescribed to this patient. Please discontinue the existing prescription to prescribe again."
```
Comment (sainak) [inline care/facility/api/serializers/prescription.py:70]:
```suggestion
                        "medicine": (
                            "This medicine is already prescribed to this patient. "
                            "Please discontinue the existing prescription to prescribe again."
                            )
```

Comment (rithviknishad) [reply care/facility/api/serializers/prescription.py:70]: Copilot had suggested me this 😅 I ignored it

**Comment by sainak** [comment]
needs test

**Review by vigneshhari — APPROVED** [review]

**Comment by vigneshhari** [comment]
I can merge once tests have been added

**Review by vigneshhari — APPROVED** [review]

**Review by sainak — APPROVED** [review]

**Pull request merged**